pkg/api/const.go

```diff
@@ -29,8 +29,9 @@ const (
 	Ubuntu1804      Distro = "ubuntu-18.04"
 	RHEL            Distro = "rhel"
 	CoreOS          Distro = "coreos"
-	AKS             Distro = "aks"
-	AKSDockerEngine Distro = "aks-docker-engine" // deprecated docker-engine distro
+	AKS             Distro = "aks"               // deprecated AKS 16.04 distro. Equivalent to aks-16.04.
+	AKSDockerEngine Distro = "aks-docker-engine" // deprecated docker-engine distro.
+	AKS1604         Distro = "aks-16.04"
```

Comment:
Suggestion: lets' change this value to "aks-ubuntu-16.04" and "aks-ubuntu-18.04. Let's assume that non-Ubuntu Linux distros may come to AKS eventually, and so the simple "16.04" string is arguably not descriptive enough.

Reply:
I agree that's better. This will break back compat though so we can't simply change the string value. If we want to do this for 18.04 we need a new distro with deprecation logic (or an alias distro that points to the same image config).

Reply:
Where's the back-compat breakage, for folks using the "aks-18.04" distro?

Reply:
if you have an existing "aks-18.04" and you try to upgrade or scale with this change, the "aks-18.04" distro won't be a valid distro and will return an error since it will have been replaced by "aks-ubuntu-18.04".

Reply:
https://github.com/Azure/aks-engine/issues/1246

Reply:
arguably it's easier and less risky to do before we move the default (since less clusters will be using the deprecated string) so doing the change in this PR might be easier, was just pointing out that this wasn't a trivial string rename.

Reply:
done